src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -351,6 +351,12 @@ public void testAvoidRedundantFetch() throws Exception {
 
         /* Without honor refspec on initial clone */
         FreeStyleProject projectWithMaster = setupProject(repos, Collections.singletonList(new BranchSpec("master")), null, false, null);
+        if (random.nextBoolean()) {
+            /* Randomly enable shallow clone, should not alter test assertions */
+            CloneOption cloneOptionMaster = new CloneOption(false, null, null);
+            cloneOptionMaster.setDepth(1);
```

Comment:
I fail to understand, why would need a shallow clone to improve coverage when we already have a full normal clone?

Reply:
Good question.  One [branch](https://github.com/jenkinsci/git-plugin/blob/c2e097dac1aa4128b0fdc77e79afe321dba76d35/src/main/java/hudson/plugins/git/GitSCM.java#L1175) was not being reached when a cloneOption extension is detected but does not have honor refspec enabled.

Reply:
Okay. I didn't consider other CloneOptions. If honor refspec is false, with avoiding the second fetch, we will also avoid other clone options like Shallow Clone, Disable tags, Reference a repo and timeout. 

The good news is that the first fetch is capable to execute all of these options if we miss the second fetch. It should not make any difference to the user's expectation on git repository information after the checkout.